clang-tidy-files-to-scan.txt

```diff
@@ -8,6 +8,11 @@
 ./iceoryx_hoofs/include/iceoryx_hoofs/cxx/deadline_timer.hpp
 ./iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp
 ./iceoryx_hoofs/include/iceoryx_hoofs/cxx/filesystem.hpp
+./iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object/*
```

Comment:
Thanks for adding the files here. But I had to refactor this code again a bit, could you please move those files into `.clang-tidy-diff-scans.txt`. You should have this file available after you rebased with the current master.

Since you also add implicitly the unchanged `shared_memory.hpp` with `./iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object/*` it is not scanned by the github CI - we perform only diff scans here -. Could you perform a local `clang-tidy-15` scan on that file. It has to be `clang-tidy-15`. But it is available on newest master in the `ice_env.sh`. Just start `./tools/scripts/ice_env.sh enter` and scan the file. You can find detailed instruction in `.clang-tidy-diff-scans.txt`.

Thanks for taking care!

Reply:
I have clang-tidy-15 integrated in my IDE via clangd and it does not show any findings.

Reply:
But the clang-tidy-15 integration just uses the clangd language server which does not analyze the file in the same depth as a manual clang-tidy-15 call. Please scan it manually.

Here are the steps:
```
./tools/scripts/ice_env.sh enter
clang-tidy-15 --warnings-as-errors=* -p build ./iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object/shared_memory.hpp
```

Reply:
Okay, I already did it manually and found nothing ... ready to merge!